dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/metadata/ServiceInstanceHostPortCustomizer.java

```diff
@@ -56,11 +60,19 @@ public void customize(ServiceInstance serviceInstance) {
                         break;
                     }
                 }
+                
+                if (host == null || port == -1) {
+                    logger.warn("The default preferredProtocol \"" + preferredProtocol + "\" is not found, fall back to the strategy that pick the first found protocol. Please try to modify the config of dubbo.application.protocol");
+                    URL url = urls.iterator().next();
+                    host = url.getHost();
+                    port = url.getPort();
+                }
             } else {
                 URL url = urls.iterator().next();
```

Comment:
It is undeterminable that which protocol will be selected

Reply:
Set does not guarantee order